AEPConcierge/Sources/Concierge+PublicAPI.swift

```diff
@@ -25,45 +25,55 @@ public extension Concierge {
     ///   - subtitle: Optional subtitle shown under the title.
     ///   - speechCapturer: Optional speech capture implementation to use.
     ///   - textSpeaker: Optional text-to-speech implementation to use.
-    static func show(
-        title: String? = nil,
-        subtitle: String? = nil,
-        speechCapturer: SpeechCapturing? = nil,
-        textSpeaker: TextSpeaking? = nil
-    ) {
-        if let speechCapturer = speechCapturer {
-            self.speechCapturer = speechCapturer
-        }
+    static func show(title: String? = nil, subtitle: String? = nil, speechCapturer: SpeechCapturing? = nil, textSpeaker: TextSpeaking? = nil) {
+        // Dispatch event to request state data necessary for displaying the UI
+        let showEvent = Event(name: Constants.EventName.SHOW_UI,
+                              type: Constants.EventType.concierge,
+                              source: EventSource.requestContent,
+                              data: nil)
+        
+        MobileCore.dispatch(event: showEvent, timeout: 3.0) { response in
+            guard let response = response, let responseData = response.data else {
+                Log.warning(label: Constants.LOG_TAG, "Unable to show chat UI - configuration is not available.")
+                return
+            }
+            
+            // get shared state, config, settings, etc. out of the response event.
+            // then show the view
+            
+            guard let config = responseData["config"] as? ConciergeConfiguration else {
```

Comment:
Can we extract this into a constant?

Reply:
updated the constant.

see comment re: dictionary vs object below :)